src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -0,0 +1,317 @@
+        textDocument,
+        parameterFilePath,
+        resourceGroupId,
+        deploymentScope,
+        "",
+        template
+      );
+    }
+  }
+
+  private async handleSubscriptionDeployment(
+    context: IActionContext,
+    textDocument: TextDocumentIdentifier,
+    documentUri: vscode.Uri,
+    deploymentScope: string,
+    template: string
+  ) {
+    const locationTreeItem =
+      await this.treeManager.azLocationTree.showTreeItemPicker<LocationTreeItem>(
+        "",
+        context
+      );
+    const location = locationTreeItem.label;
+    const subscriptionId = locationTreeItem.subscription.subscriptionPath;
+    const parameterFilePath = await this.selectParameterFile(
+      context,
+      documentUri
+    );
+
+    await this.sendDeployCommand(
+      textDocument,
+      parameterFilePath,
+      subscriptionId,
+      deploymentScope,
+      location,
+      template
+    );
+  }
+
+  private async sendDeployCommand(
+    textDocument: TextDocumentIdentifier,
+    parameterFilePath: string,
+    id: string,
+    deploymentScope: string,
+    location: string,
+    template: string
+  ) {
+    const bicepDeployParams: BicepDeployParams = {
+      textDocument,
+      parameterFilePath,
+      id,
+      deploymentScope,
+      location,
+      template,
+    };
+    const deploymentResponse: string = await this.client.sendRequest(
+      bicepDeployRequestType,
+      bicepDeployParams
+    );
+
+    this.outputChannelManager.appendToOutputChannel(deploymentResponse);
+  }
+
+  private async selectParameterFile(
+    _context: IActionContext,
+    sourceUri: Uri | undefined
+  ): Promise<string> {
+    const quickPickItems: IAzureQuickPickItem[] =
+      await this.createParameterFileQuickPickList();
+    const result: IAzureQuickPickItem = await _context.ui.showQuickPick(
+      quickPickItems,
+      {
+        canPickMany: false,
+        placeHolder: `Select a parameter file`,
+        suppressPersistence: true,
+      }
+    );
+
+    if (result.label.includes("Browse...")) {
+      const paramsPaths: Uri[] | undefined = await vscode.window.showOpenDialog(
+        {
+          canSelectMany: false,
+          defaultUri: sourceUri,
+          openLabel: "Select Parameter File",
+        }
+      );
+      if (paramsPaths && paramsPaths.length == 1) {
+        const parameterFilePath = paramsPaths[0].fsPath;
+        this.outputChannelManager.appendToOutputChannel(
+          `Parameter file used in deployment -> ${path.basename(
+            parameterFilePath
+          )}`
+        );
+        return parameterFilePath;
+      }
+    }
+
+    this.outputChannelManager.appendToOutputChannel(
+      `Parameter file was not provided`
+    );
+
+    return "";
+  }
+
+  private async createParameterFileQuickPickList(): Promise<
+    IAzureQuickPickItem[]
+  > {
+    const none: IAzureQuickPickItem = {
+      label: "$(circle-slash) None",
+      data: undefined,
+    };
+    const browse: IAzureQuickPickItem = {
```

Comment:
Did we decide not to show them a list of params files in the bicep file's folder?

Reply:
We decided to present the user with all of the .json files (besides the .json of the exact same name as the .bicep file). Here's the work item I logged after our discussion:
https://github.com/Azure/bicep/issues/6155

Let me know if you have any questions/suggestions. Thanks!